sklearn/neighbors/binary_tree.pxi

```diff
@@ -654,6 +654,56 @@ cdef class NeighborsHeap:
 
         return 0
 
+    cdef int cpush(self, ITYPE_t row, DTYPE_t val,
```

Comment:
Nice advise. 

`push` is used in tests of tree structures. I also will make it simply `def` and replace `push` to `cpush` everywhere in pyx-files.